stan/math/prim/fun/eigen_comparisons.hpp

```diff
@@ -0,0 +1,39 @@
+#ifndef STAN_MATH_PRIM_FUN_EIGEN_COMPARISONS_HPP
+#define STAN_MATH_PRIM_FUN_EIGEN_COMPARISONS_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <stan/math/prim/fun/value_of.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Add support for comparisons involving Eigen types with different scalars,
+ * where one of the scalars is an autodiff type. This includes comparisons of
+ * an Eigen type and a scalar.
+ * @param OPERATOR name of the operator function to implement
+ * @param OP operator to use for comparison of values
+ **/
+#define ADD_MIXED_AUTODIFF_SCALAR_COMPARISON(OPERATOR, OP) \
```

Comment:
This needs some tests.

I assume two tests for each operation. One with vars on one side and one on the other. Compare the results against just comparing two Eigen plain types of values.